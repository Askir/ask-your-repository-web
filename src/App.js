import React, { Component } from 'react';
import './App.scss';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <p>
            HP2018HG1 - Website
          </p>
        </header>
      </div>
    );
  }
}

export default App;
