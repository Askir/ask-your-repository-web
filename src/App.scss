.App {
  max-width: 1200px;
  height: 100%;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
}
