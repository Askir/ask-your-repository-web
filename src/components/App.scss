@import '../style/breakpoints.scss';

.App {
  height: 100%;

  &__inner {
    height: 100%;
    padding: 0 var(--padding-sides) 0px var(--padding-sides);
    padding-top: var(--navbar-height);

    display: flex;
    align-items: flex-start;
    justify-content: center;

    @media (min-width: $xs-to-sm) {
      transition: margin-left var(--sidebar-animation-duration);
      &--with-sidebar { margin-left: var(--team-sidebar-width); }
    }
  }
}
