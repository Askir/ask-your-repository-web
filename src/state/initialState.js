// @flow

// This file is only for state related testing

import type { AppState } from './AppState';
import { initialState as presentation } from './presentation/presentation.reducer';
import { initialState as teamSidebar } from './team_sidebar/teamSidebar.reducer';
import { initialState as activeTeam } from './active_team/activeTeam.reducer';
import { initialState as image } from './image/image.reducer';

const initialState: AppState = {
  flash: { messages: [] },
  presentation,
  image,
  activeTeam,
  teamSidebar,
};

export default initialState;
