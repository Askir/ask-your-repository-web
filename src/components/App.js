// @flow
import React from 'react';
import { Route, Switch } from 'react-router';
import { connect } from 'react-redux';
import LoginForm from './auth/LoginForm';
import RegisterForm from './auth/RegisterForm';
import PasswordRequestForm from './auth/forgot_password/PasswordRequestForm';
import ResetPasswordForm from './auth/forgot_password/ResetPasswordForm';
import LandingPage from './landing_page/LandingPage';
import FlashMessages from './utility/flash/FlashMessages';
import AuthorizedApp from './AuthorizedApp';
import Footer from './footer/Footer';
import type { AppState } from '../state/AppState';
import './App.scss';

type Props = {
  isAuthenticated: boolean,
};

function App(props: Props) {
  return (
    <div className="App">
      <FlashMessages />
      <Switch>
        <Route path="/login" component={LoginForm} />
        <Route path="/register" component={RegisterForm} />
        <Route path="/forgot-password" component={PasswordRequestForm} />
        <Route path="/reset-password" component={ResetPasswordForm} />
        <Route path="/" component={props.isAuthenticated ? AuthorizedApp : LandingPage} />
      </Switch>
      <Footer />
    </div>
  );
}

const mapStateToProps = (state: AppState) => ({
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(mapStateToProps)(App);
