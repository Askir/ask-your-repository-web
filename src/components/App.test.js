// @flow
import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import initialState from '../state/initialState';
import App from './App';

const mockStore = configureMockStore();

describe('<App />', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<App store={store} />).dive();
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
