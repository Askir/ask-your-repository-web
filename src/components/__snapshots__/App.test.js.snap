// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <Connect(FlashMessages) />
  <Connect(OnClickOutside(TeamSidebar)) />
  <Connect(NavBar) />
  <div
    className="App__inner App__inner--with-sidebar"
  >
    <Switch>
      <Redirect
        exact={true}
        from="/"
        push={false}
        to="/images"
      />
      <Route
        component={[Function]}
        path="/upload"
      />
      <Route
        component={[Function]}
        path="/tagging"
      />
      <Route
        component={[Function]}
        path="/images"
      />
      <Route
        component={[Function]}
        path="/presentation"
      />
    </Switch>
  </div>
</div>
`;
